### PR TITLE
feat: sharky disaster recovery

### PR DESCRIPTION
**Pull request by notanatol**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md)
- [ ] My change requires a documentation update and I have done it
- [ ] I have added tests to cover my changes.

### Description
Localstore must be able to recover after a "dirty" shutdown, avoiding data corruption.

### Related Issue
[sharky DRP](https://github.com/ethersphere/bee-backlog/issues/12)

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2722)

### CODE REVIEW

File: pkg/localstore/disaster_recovery.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
+// Copyright 2021 The go-ethereum Authors
```
Comment (mrekucci) [inline pkg/localstore/disaster_recovery.go:1]: s the header still relevant (also in other files)?

File: pkg/localstore/disaster_recovery.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+			e.Location = value[headerSize:]
+			return e, nil
+		},
+	})
+
+	// this operation is memory intensive. we will preallocate the
+	// minimum size of locations we expect to see.
+	usedLocations = make(chan locOrErr)
+
+	go func() {
```
Comment (mrekucci) [inline pkg/localstore/disaster_recovery.go:82]: There is no cancelation story for this goroutine, the `db.ctx` can be leverage here.

File: pkg/localstore/localstore.go [hunk 3]
```diff
@@ -749,3 +788,27 @@ func totalTimeMetric(metric prometheus.Counter, start time.Time) {
 	metric.Add(float64(totalTime))
 
 }
+
+func isDirtyShutdown(path string) bool {
+	_, err := os.Stat(filepath.Join(path, sharkyDirtyFileName))
+	return !os.IsNotExist(err)
```
Comment (mrekucci) [inline pkg/localstore/localstore.go:794]: This function predates `errors.Is`. It only supports errors returned by the os package. New code should use `errors.Is(err, fs.ErrNotExist)`.

File: pkg/localstore/disaster_recovery.go (hunk 1 above)
Comment (notanatol) [reply pkg/localstore/disaster_recovery.go:1]: not sure, @acud ?

File: pkg/localstore/disaster_recovery.go (hunk 2 above)
Comment (notanatol) [reply pkg/localstore/disaster_recovery.go:84]: fixed

File: pkg/localstore/localstore.go (hunk 3 above)
Comment (notanatol) [reply pkg/localstore/localstore.go]: fixed

**Pull request closed without merging**